pkg/virt-handler/cache/cache.go

```diff
@@ -76,66 +75,68 @@ type ghostRecord struct {
 	UID        types.UID `json:"uid"`
 }
 
-var ghostRecordGlobalCache map[string]ghostRecord
-var ghostRecordGlobalMutex sync.Mutex
-var checkpointManager IterableCheckpointManager
-
-func InitializeGhostRecordCache(directoryPath string) error {
-	ghostRecordGlobalMutex.Lock()
-	defer ghostRecordGlobalMutex.Unlock()
+type GhostRecordStore struct {
+	cache             map[string]ghostRecord
+	mutex             sync.Mutex
+	checkpointManager checkpoint.CheckpointManager
+}
 
-	ghostRecordGlobalCache = make(map[string]ghostRecord)
+func InitializeGhostRecordCache(directoryPath string) (*GhostRecordStore, error) {
+	iterablecpm := newIterableCheckpointManager(directoryPath)
```

Comment:
To be 100% certain:
the elimination of the lock is voluntary, right?

Reply:
Well the lock is needed only if you call this in multiple goroutines/threads...  We don't call this in the way and transforming this away from singleton will also drop the requirement to have the lock